modules/costs/app/controllers/costlog_controller.rb

```diff
@@ -115,22 +115,6 @@ def find_project
     render_404
   end
 
-  def find_optional_project
-    if params[:work_package_id].present?
-      @work_package = WorkPackage.find(params[:work_package_id])
-      @project = @work_package.project
-    elsif params[:work_package_id].present?
-      @work_package = WorkPackage.find(params[:work_package_id])
-      @project = @work_package.project
-    elsif params[:project_id].present?
-      @project = Project.find(params[:project_id])
-    end
-
-    if params[:cost_type_id].present?
-      @cost_type = CostType.find(params[:cost_type_id])
-    end
-  end
```

Comment:
This method is different from the other `find_optional_project` methods. Are we sure we never encountr the case where we have `params[:work_package_id]` set?

Reply:
In this case I don't care since the method AFAIK is not called 😆 . It is in the private section and no call to `find_optional_project` is in the file.